c7n/resources/asg.py

```diff
@@ -286,7 +286,10 @@ def get_asg_errors(self, asg):
         for bd in cfg['BlockDeviceMappings']:
             if 'Ebs' not in bd or 'SnapshotId' not in bd['Ebs']:
                 continue
-            if bd['Ebs']['SnapshotId'] not in self.snapshots:
+            snapshot_id = bd['Ebs']['SnapshotId']
+            if hasattr(snapshot_id, 'strip'):
```

Comment:
yeah - was just being defensive.  I can remove that if you feel confident it will always be a string.